src/controller.ts

```diff
@@ -0,0 +1,49 @@
+import { Component } from "./component.ts";
+
+export abstract class Controller {
+  private static controllerMap: Map<string, Controller> = new Map();
+
+  protected readonly components: Component[] = [];
+  protected readonly componentsByEvents: Map<string, Component[]> = new Map();
+
+  register(component: Component) {
+    this.components.push(component);
+  }
+
+  registerByEvent(event: string, component: Component) {
```

Comment:
what does this do?

Reply:
This is likely the main way to subscribe to events. You are registering this component as interested in those events.

I should probably switch it to subscribe, and if you didn't think it was clear, that is all the more reason to.